src/category_theory/glue_data.lean

```diff
@@ -0,0 +1,201 @@
+
+attribute [simp] glue_data.t_id
+attribute [instance] glue_data.f_id glue_data.f_mono glue_data.f_has_pullback
+attribute [reassoc] glue_data.t_fac glue_data.cocycle
+
+namespace glue_data
+
+variables {C} (D : glue_data C)
+
+@[simp] lemma t'_iij (i j : D.ι) : D.t' i i j = (pullback_symmetry _ _).hom :=
+begin
+  have eq₁ := D.t_fac i i j,
+  have eq₂ := (is_iso.eq_comp_inv (D.f i i)).mpr (@pullback.condition _ _ _ _ _ _ (D.f i j) _),
+  rw [D.t_id, category.comp_id, eq₂] at eq₁,
+  have eq₃ := (is_iso.eq_comp_inv (D.f i i)).mp eq₁,
+  rw [category.assoc, ←pullback.condition, ←category.assoc] at eq₃,
+  exact mono.right_cancellation _ _
+    ((mono.right_cancellation _ _ eq₃).trans (pullback_symmetry_hom_comp_fst _ _).symm)
+end
+
+lemma t'_jii (i j : D.ι) : D.t' j i i = pullback.fst ≫ D.t j i ≫ inv pullback.snd :=
+by { rw [←category.assoc, ←D.t_fac], simp }
+
+lemma t'_iji (i j : D.ι) : D.t' i j i = pullback.fst ≫ D.t i j ≫ inv pullback.snd :=
+by { rw [←category.assoc, ←D.t_fac], simp }
+
+@[simp, reassoc, elementwise] lemma t_inv (i j : D.ι) :
+  D.t i j ≫ D.t j i = 𝟙 _ :=
+begin
+  have eq : (pullback_symmetry (D.f i i) (D.f i j)).hom = pullback.snd ≫ inv pullback.fst,
+  { simp },
+  have := D.cocycle i j i,
+  rw [D.t'_iij, D.t'_jii, D.t'_iji, fst_eq_snd_of_mono_eq, eq] at this,
+  simp only [category.assoc, is_iso.inv_hom_id_assoc] at this,
+  rw [←is_iso.eq_inv_comp, ←category.assoc, is_iso.comp_inv_eq] at this,
+  simpa using this,
+end
+
+lemma t'_inv (i j k : D.ι) : D.t' i j k ≫ (pullback_symmetry _ _).hom ≫
+  D.t' j i k ≫ (pullback_symmetry _ _).hom = 𝟙 _ :=
+begin
+  rw ← cancel_mono (pullback.fst : pullback (D.f i j) (D.f i k) ⟶ _),
+  simp [t_fac, t_fac_assoc]
+end
+
+instance t_is_iso (i j : D.ι) : is_iso (D.t i j) :=
+⟨⟨D.t j i, D.t_inv _ _, D.t_inv _ _⟩⟩
+
+instance t'_is_iso (i j k : D.ι) : is_iso (D.t' i j k) :=
+⟨⟨D.t' j k i ≫ D.t' k i j, D.cocycle _ _ _, (by simpa using D.cocycle _ _ _)⟩⟩
+
+@[reassoc]
+lemma t'_comp_eq_pullback_symmetry (i j k : D.ι) :
+  D.t' j k i ≫ D.t' k i j = (pullback_symmetry _ _).hom ≫
+  D.t' j i k ≫ (pullback_symmetry _ _).hom :=
+begin
+  transitivity inv (D.t' i j k),
+  { exact is_iso.eq_inv_of_hom_inv_id (D.cocycle _ _ _) },
+  { rw ← cancel_mono (pullback.fst : pullback (D.f i j) (D.f i k) ⟶ _),
+    simp [t_fac, t_fac_assoc] }
+end
+
+/-- (Implementation) The disjoint union of `U i`. -/
+def sigma_opens [has_coproduct D.U] : C := ∐ D.U
+
+/-- (Implementation) The diagram to take colimit of. -/
+def diagram : multispan_index C :=
+{ L := D.ι × D.ι, R := D.ι,
+  fst_from := _root_.prod.fst, snd_from := _root_.prod.snd,
+  left := D.V, right := D.U,
+  fst := λ ⟨i, j⟩, D.f i j,
+  snd := λ ⟨i, j⟩, D.t i j ≫ D.f j i }
+
+section
+
+variable [has_multicoequalizer D.diagram]
+
+/-- The glued object given a family of gluing data. -/
+def glued : C := multicoequalizer D.diagram
+
+/-- The map `D.U i ⟶ D.glued` for each `i`. -/
+def imm (i : D.ι) : D.U i ⟶ D.glued :=
```

Comment:
Would it make sense to call this `ι`?

Reply:
This would probably clash with the `ι` used for the index set.
Or should I change the naming of the index set and keep `ι` for this map.

Reply:
Hmm, I like that idea. Maybe use `I` or `J` or something like that for the indexing set. Because `ι` is used a lot for the canonical inclusions into some object in the limits library.